### PR TITLE
test(flask): run flask snapshot tests with werkzeug instead of uwsgi

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Using uwsgi in the flask client results in [flaky snapshot tests](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/20152/workflows/0e7c9f4e-80cb-4f3a-ab5c-1ea63afa7d8f/jobs/1364413). Instead of investigating and fixing these failures this PR replaces uwsgi with the default flask development server (werkzeug).

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] PR cannot be broken up into smaller PRs.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: tests/contrib/flask/test_flask_snapshot.py [hunk 1]
```diff
@@ -87,9 +83,9 @@ def flask_client(flask_command, flask_env, flask_port):
         # but the test agent hasn't necessarily finished processing
         # the traces (race condition) so wait just a bit for that
         # processing to complete.
-        time.sleep(0.2)
+        time.sleep(0.5)
```
Comment (mabdinur) [inline tests/contrib/flask/test_flask_snapshot.py:86]:
```suggestion
        time.sleep(0.2)
```

File: tests/contrib/flask/test_flask_snapshot.py [hunk 2]
```diff
@@ -87,9 +83,9 @@ def flask_client(flask_command, flask_env, flask_port):
...
     finally:
-        os.killpg(proc.pid, signal.SIGKILL)
+        os.kill(proc.pid, signal.SIGKILL)
```
Comment (mabdinur) [inline tests/contrib/flask/test_flask_snapshot.py:88]:
```suggestion
        os.killpg(proc.pid, signal.SIGKILL)
```

Comment (Kyle-Verhoog) [inline tests/contrib/flask/test_flask_snapshot.py:88]:
```suggestion
        os.killpg(proc.pid, signal.SIGKILL)
```

File: tests/contrib/flask/test_flask_snapshot.py (hunk 1 above)
Comment (mabdinur) [inline tests/contrib/flask/test_flask_snapshot.py:86]:
```suggestion
        time.sleep(0.2)
```

File: tests/contrib/flask/test_flask_snapshot.py (hunk 2 above)
Comment (mabdinur) [inline tests/contrib/flask/test_flask_snapshot.py:88]:
```suggestion
        os.killpg(proc.pid, signal.SIGKILL)
```

**Review by Kyle-Verhoog — APPROVED** [review]
uwsgi....

**Review by ZStriker19 — APPROVED** [review]

**Comment by mabdinur** [comment]
@Mergifyio backport 1.4 1.3 0.61

**Pull request merged**